napalm/base/__init__.py

```diff
@@ -67,16 +67,16 @@ def get_network_driver(module_name, prepend=True):
         napalm.base.exceptions.ModuleImportError: Cannot import "napalm_wrong". Is the library \
         installed?
     """
-    if module_name == "mock":
+    if name == "mock":
         return MockDriver
 
-    if not (isinstance(module_name, py23_compat.string_types) and len(module_name) > 0):
+    if not (isinstance(name, py23_compat.string_types) and len(name) > 0):
```

Comment:
This looks like a bug...not from your change, but in the original code. I think it was supposed to be a logical or (i.e. not a string or the string is null).

Reply:
I just tested it and it seems to work fine:

```
>>> from napalm import get_network_driver
>>> get_network_driver("")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "napalm/base/__init__.py", line 74, in get_network_driver
    raise ModuleImportError('Please provide a valid driver name.')
napalm.base.exceptions.ModuleImportError: Please provide a valid driver name.
```

so I don't think it's a bug. Note that the not in the beginning basically inverts the logical operand. So that's basically equivalent to:

```
not isinstance(name, py23_compat.string_types or not len(name):
```

Reply:
Yeah, misread it because of how they grouped the parenthesis and `not` statement...so no issue.